sklearn/linear_model/ridge.py

```diff
@@ -867,10 +871,12 @@ def fit(self, X, y, sample_weight=None):
             #fit_params = {'sample_weight' : sample_weight}
             fit_params = {}
             gs = GridSearchCV(Ridge(fit_intercept=self.fit_intercept),
-                              parameters, fit_params=fit_params, cv=self.cv)
+                              parameters, fit_params=fit_params, cv=self.cv,
+                              scoring = self.scoring)
```

Comment:
The current regression test already tests if the scoring functions is taken into account (through reg3 and reg4 comparison of `self.best_score_`).
To fail on master branch, the test should test it through `self.alpha_`, but it is not easy to find an example which gives 2 different alphas for 2 different scoring functions.